Fix tests

sevein opened the pull request:
I had not realized that we were not running Travis CI in this repository. It's been very important for us in vanilla. Since we created this fork vanilla received a number of improvements and fixes in its test suite. This pull request brings the related commits.

**Please read the description of the commits for more details.**

cole committed:
Added migration for locations app i18n changes

[ Added by @sevein ]: during the i18n work we did in the Storage Service
we made some changes to our Django models but we forgot to commit the
corresponding database migrations. The missing migration is added in this
commit - however I don't think it actually affects the schema of the database
but it's still a required step because Django needs any model change to be
recorded as a migration. It's important to add missing migrations like this
because we're planning to add a check in CI for this purpose.

sevein committed:
Only require ALLOWED_HOSTS AND SECRET_KEY in prod

This saves us from having to define these environments when not needed, e.g.
when we're running our tests or we're in a development environment. In other
words this simplifies the configuration of non-production environments.

sevein committed:
Docker: allow tests to run rootless

This commit updates `TestDuracloud` so any temporary dir/file is created in the
SS internal working directory instead of mutating the source code tree. This is
important because in our `Dockerfile` the source tree isn't mutable causing the
tests to fail unless they were executed as the `root` user.

sevein committed:
Stop looking up DB envs in base settings module

This commit moves database configuration environment lookups to the production
(`production.py`) and development (`local.py`) settings module. Before, the
lookups were in `base.py` which also affected `test.py`. By making this change
we simplify the configuration of the test environment by not requiring to define
database configuration parameters as we prefer to default to SQLite's `:memory`
backend.

sevein committed:
Tests: add admin user to fixtures

Storage Service used to automatically add a `test` user to the database. This
behaviour was removed in JiscRDSS/SS for security reasons but it broke our
tests because they rely on that user. In this commit we make the user available
again for the tests by adding it to the fixtures - this won't affect
deployments.

lower29 reviewed (changes requested):
This all looks okay but I must admit I had difficulty tracking why some of the changes were necessary to "fix the tests". Would be good to have more explanation, probably in the commit messages, or as in-line comments on here, as to why a particular change was made and what test it allowed to pass.

I apologise that I'm not very familiar with the codebase; some of it is probably obvious to those who are.

sevein committed:
Fix linting issue

sevein committed:
Dockerfile: pip install in quiet mode

The default is very verbose and unnecessary.

sevein committed:
Tests: check for migrations

sevein committed:
Travis: small fixes and improvements

- Run in containers because it's faster (`sudo: false`).
- Install OS dependencies using the `addons` syntax.
- Install missing OS dependencies because tests were breaking.
- Install `tox` in `before_install` (so it's done only once).
- Use `install: true` so Travis does not install requirements automatically
since that's handled in `tox.ini`.
- Make `/var/archivematica/storage_service` writeable so tests can run.

sevein commented:
@lower29 please take another look when you have a chance! Thanks.

lower29 commented:
Much clearer with the improved commit messages, thank you! Approved 👍
